Microsoft.Toolkit.Uwp.UI/Extensions/TextBoxRegEx/TextBoxRegex.cs

```diff
@@ -102,22 +104,18 @@ private static void ValidateTextBox(TextBox textbox, bool force = true)
             if (!regexMatch && force)
             {
                 var validationModel = (ValidationMode)textbox.GetValue(ValidationModeProperty);
-                if (validationModel == ValidationMode.Forced)
+                if (validationModel == ValidationMode.Forced || validationModel == ValidationMode.Instantly)
                 {
-                    if (!string.IsNullOrEmpty(textbox.Text))
+                    textbox.Text = string.Empty;
+                }
+                if (!string.IsNullOrEmpty(textbox.Text))
+                {
+                    if (validationModel == ValidationMode.Dynamic)
                     {
-                        var validationModel = (ValidationMode)textbox.GetValue(ValidationModeProperty);
-                        if (validationModel == ValidationMode.Forced || validationModel == ValidationMode.Instantly)
-                        {
-                            textbox.Text = string.Empty;
-                        }
-                        else if (validationModel == ValidationMode.Dynamic)
+                        textbox.Text = textbox.Text.Remove(textbox.Text.Length - 1);
```

Comment:
You should use the textbox selection info in order to erase the character at its insertion point, it's a bit odd for the rest of the data to be axed if the user inserts something not at the end.

Reply:
Done. But if I insert a string which is invalid, for example "34234f34234", the text will be deleted up to "34234".